crates/polars-pipe/src/executors/sinks/ordered.rs

```diff
@@ -57,32 +55,7 @@ impl Sink for OrderedSink {
         self.sort();
 
         let chunks = std::mem::take(&mut self.chunks);
-        let mut combiner = StreamingVstacker::default();
```

Comment:
If we would rechunk, we could simply rechunk here. But I don't want to do that as that should be left to the consumer of the streaming engine.

Reply:
In that case it should probably be done in `write_parquet()`, otherwise the `collect(streaming=True).write_parquet()` case will continue to be slow.

Reply:
Which would requite the new struct be e.g. moved into the `polars-core` crate and made public.

Here's the runtime with latest commit on my computer, last case is slow again:

```
SINK 1.0721302032470703
COLLECT+WRITE 0.9235994815826416
STREAMING COLLECT+WRITE 4.254129648208618
```

Reply:
But maybe also `write_feather()`. or the cloud parquet writer. etc.. (Having it in OrderedSink seemed like a low-cost smoothing of performance bumps, limited to a single place.)

Reply:
Then that logic should be in `write_parquet` indeed. That writer should check the chunk sizes.

I will first merge this one, and then we can follow up with the `write_parquet` optimization.

Reply:
> But maybe also `write_feather()`. or the cloud parquet writer. etc.. (Having it in OrderedSink seemed like a low-cost smoothing of performance bumps, limited to a single place.)

Yes, but it is more expensive for other operations. Operations themselves should knie their best chunking strategies.